examples/DotNetCore/ConsoleAppWithTableStorageReference/Program.cs

```diff
@@ -0,0 +1,161 @@
+                            {
+                                // Example value for key "MyShop:Inventory" in App Configuration: https://{account_name}.table.core.windows.net/{table_name}
+                                string tableContent = await ReadTableContentAsync(new Uri(setting.Value));
+
+                                setting = new ConfigurationSetting(setting.Key, tableContent, setting.Label);
+                                setting.ContentType = "application/json";
+                            }
+
+                            return setting;
+                        });
+            });
+
+            Configuration = builder.Build();
+        }
+
+        private static StringBuilder BuildProductTable()
+        {
+            StringBuilder sb = new StringBuilder();
+
+            List<Product> tableContent = new List<Product>();
+            Configuration.GetSection("MyShop:Inventory").Bind(tableContent);
+
+            // Example value for "MyShop:DisplayedColumns": [Name, OnSale]
+            // Content-Type: application/json
+            List<string> columnsToDisplay = new List<string>();
+            Configuration.GetSection("MyShop:DisplayedColumns").Bind(columnsToDisplay);
+
+            foreach (Product product in tableContent)
+            {
+                if (columnsToDisplay.Contains(nameof(product.Name)))
+                {
+                    sb.AppendLine($"Name = {product.Name}");
+                }
+
+                if (columnsToDisplay.Contains(nameof(product.Quantity)))
+                {
+                    sb.AppendLine($"Quantity = {product.Quantity.ToString()}");
+                }
+
+                if (columnsToDisplay.Contains(nameof(product.OnSale)))
+                {
+                    sb.AppendLine($"OnSale = {product.OnSale.ToString()}");
+                }
+
+                sb.AppendLine();
+            }
+
+            return sb;
+        }
+
+        private static async Task<string> ReadTableContentAsync(Uri tableUri)
+        {
+            string[] pathSegments = tableUri.Segments;
+
+            // The last segment in the path should be the table name
+            string tableName = pathSegments[pathSegments.Length - 1];
+
+            var tableClient = new TableClient(tableUri, tableName, new DefaultAzureCredential());
+
+            // Adding example products to display
+            var prod1 = new Product()
+            {
+                RowKey = "68719518388",
+                PartitionKey = "gear-surf-surfboards",
+                Name = "Ocean Surfboard",
+                Quantity = 8,
+                OnSale = true
+            };
+
+            var prod2 = new Product()
+            {
+                RowKey = "68719518390",
+                PartitionKey = "gear-surf-surfboards",
+                Name = "Sand Surfboard",
+                Quantity = 5,
+                OnSale = false
+            };
+
+            await tableClient.UpsertEntityAsync(prod1);
```

Comment:
This is performing a Write? Doesn't match the method name.

Reply:
Yea wanted to have some data in the table by default, I could change the method name to reflect that.

Reply:
How about writing the data in the beginning of the application.

Reply:
I'm not sure where that would be, since we actually get the Uri when it's pulled from app config.

Reply:
Good point. Needs some thought, but writing it in the exact same code path as reading it is weird.

Reply:
Or, don't worry about writing data in the example. I don't believe our other examples do such a thing.